docs/design/proposals/multi-version/multi-version.md

```diff
@@ -0,0 +1,259 @@
+# ACK Multi-Version CRD support
+
+## Background
+
+### ACK and upgrade issues
+
+AWS Controllers for Kubernetes (ACK) is a set of open-source controllers for
+managing AWS resources in Kubernetes. It's designed to simplify the management
+of AWS resources in Kubernetes by taking advantage of the native kubernetes
+management system known as the Control loop or Reconcilliating systems.
+
+Currently, ACK uses a code-genrator to generate everything that is need to
+build and run a controller: API definitions, CRDs, AWSResourceManager, helm
+charts etc... 98% of the controllers code is generated by the ACK community
+members. When a new service team introduces a new field, resource or
+operation all we do on the ACK side is bumping the aws-sdk-go version and
+regenerate the controller. :magic:
+
+However, when service teams/ack developers have to rename a field or make
+a field a kubernetes secret reference, they make significant changes to
+the CRD that are backward incompatible. In kubernetes world, CRD schemas
+should not be modified but rather introduced in a new CRD version.
+
+### Kubernetes and multi-version CRDs
+
+In a multi-version CRDs, each version of the resource is identified by a
+distinct version string (v1alpha1, v1beta2, v1 etc...). Kubernetes uses the
+version string to route API requests to the appropriate version of the
+resource.
+
+Even when dealing with multiple versions, we still need one controller handling
```

Comment:
For readers not deeply familiar with multi-versioning, this would be a good place to provide a link to existing reading on concepts like Hubs and Spokes.  [Here is one](https://book.kubebuilder.io/multiversion-tutorial/conversion-concepts.html) that I found, feel free to use a different one.